x-pack/solutions/observability/plugins/streams/server/lib/streams/helpers/lifecycle.test.ts

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { WiredStreamDefinition } from '@kbn/streams-schema';
+import { findInheritedLifecycle, findInheritingStreams } from './lifecycle';
+
+describe(__filename, () => {
+  describe('findInheritedLifecycle', () => {
+    describe('picks the definition lifecycle', () => {
+      const definition = {
+        name: 'one.two',
+        ingest: { lifecycle: { type: 'dlm', data_retention: '1d' } },
+      } as WiredStreamDefinition;
+      const ascendants = [
```

Comment:
Language nit: Ancestor is the more commonly used word (in general and in the code base) though I can see the symmetry in ascendant and descendant 😅